app/src/main/java/com/parking_system_kotlin/mvp/presenter/ParkingReservationPresenter.kt

```diff
@@ -0,0 +1,28 @@
+package com.parking_system_kotlin.mvp.presenter
+
+import com.parking_system_kotlin.listeners.ListenerDateTime
+import com.parking_system_kotlin.mvp.contracts.ParkingReservationContract
+import com.parking_system_kotlin.utils.DateUtils
+import java.util.Calendar
+
+class ParkingReservationPresenter(private val view: ParkingReservationContract.ParkingReservationView) :
+    ParkingReservationContract.ParkingReservationPresenter {
+
+    override fun showDatePicker(listenerDateTime: ListenerDateTime, dateSelector: Boolean) {
+        view.showDatePicker(listenerDateTime, dateSelector)
+    }
+
+    override fun setEntryDate(entryDate: Calendar) {
+        val stringDate: String = DateUtils().getStringFromDate(entryDate)
```

Comment:
As a future recommendation instead to create a DateUtils class to do this transformation of date, you can create a extension function, so the idea is that you do this: "entryDate.getStringFromDate()" you use the date an automatically transform to String. This is basically to avoid a creation of an extra class.

Reply:
nice! applied the recommendation. thanks!